src/main/java/org/wise/portal/presentation/web/controllers/student/StudentAPIController.java

```diff
@@ -407,8 +407,17 @@ HashMap<String, Object> canBeAddedToWorkgroup(Authentication auth,
     Workgroup workgroup = null;
     if (workgroupId != null) {
       workgroup = workgroupService.retrieveById(workgroupId);
-    } else if (workgroupService.isUserInAnyWorkgroupForRun(user, run)) {
-      workgroup = workgroupService.getWorkgroupListByRunAndUser(run, user).get(0);
+    }
+    Workgroup workgroupOfUserBeingAdded = null;
+    if (workgroupService.isUserInAnyWorkgroupForRun(user, run)) {
+      workgroupOfUserBeingAdded = workgroupService.getWorkgroupListByRunAndUser(run, user).get(0);
+      if (isUserAlreadyInAnotherWorkgroup(workgroup, workgroupOfUserBeingAdded)) {
+        response.put("status", false);
+        return response;
+      }
+    }
+    if (workgroup == null) {
+      workgroup = workgroupOfUserBeingAdded;
```

Comment:
The logic in this block is a bit hard to follow. Is there a cleaner way to write this? Would something like this work?
```suggestion
    if (workgroupService.isUserInAnyWorkgroupForRun(user, run)) {
      Workgroup workgroupOfUserBeingAdded = workgroupService.getWorkgroupListByRunAndUser(run, user).get(0);
      if (isUserAlreadyInAnotherWorkgroup(workgroup, workgroupOfUserBeingAdded)) {
        response.put("status", false);
        return response;
      } else {
        workgroup = workgroupOfUserBeingAdded;
      }
    }
```